network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/WgVpnNetworkStack.kt

```diff
@@ -53,56 +49,44 @@ class WgVpnNetworkStack @Inject constructor(
     private val networkProtectionRepository: Lazy<NetworkProtectionRepository>,
     private val currentTimeProvider: CurrentTimeProvider,
     private val netpPixels: Lazy<NetworkProtectionPixels>,
-    private val netPDefaultConfigProvider: NetPDefaultConfigProvider,
+    // private val netPDefaultConfigProvider: NetPDefaultConfigProvider,
     private val dnsProvider: DnsProvider,
     private val crashLogger: CrashLogger,
 ) : VpnNetworkStack {
-    private var wgTunnelData: WgTunnelData? = null
+    // private var wgTunnelData: WgTunnelData? = null
+    private var wgConfig: Config? = null
 
     override val name: String = NetPVpnFeature.NETP_VPN.featureName
 
     override fun onCreateVpn(): Result<Unit> = Result.success(Unit)
 
     override suspend fun onPrepareVpn(): Result<VpnTunnelConfig> {
-        fun WgTunnelData.allDns(): Set<InetAddress> {
-            return mutableSetOf<InetAddress>().apply {
-                runCatching { InetAddress.getByName(gateway) }.getOrNull()?.let {
-                    add(it)
-                }
-                addAll(netPDefaultConfigProvider.fallbackDns())
-            }.toSet().also { logcat { "DNS to be configured $it" } }
+        suspend fun fetchWireguardConfigAndCacheIt(): Config {
+            logcat { "Fetching new wireguard config" }
+            return wgTunnelLazy.get().establish()
+                .onFailure { netpPixels.get().reportErrorInRegistration() }
+                .getOrThrow()
         }
-
         return try {
             netpPixels.get().reportEnableAttempt()
 
-            wgTunnelData = wgTunnelLazy.get().establish()
-                .onFailure { netpPixels.get().reportErrorInRegistration() }
-                .getOrThrow()
-            logcat { "Received config from BE: $wgTunnelData" }
-
-            networkProtectionRepository.get().run {
-                serverDetails = ServerDetails(
-                    serverName = wgTunnelData!!.serverName,
-                    ipAddress = wgTunnelData!!.serverIP,
-                    location = wgTunnelData!!.serverLocation,
-                )
-                clientInterface = ClientInterface(
-                    tunnelCidrSet = wgTunnelData!!.tunnelAddress.toCidrString(),
-                )
-            }
+            wgConfig = networkProtectionRepository.get().wireguardConfig ?: fetchWireguardConfigAndCacheIt()
+            logcat { "Wireguard configuration:\n$wgConfig" }
+
+            // refresh WG config
+            networkProtectionRepository.get().wireguardConfig = wgConfig
```

Comment:
NIT:  we only do this in fetchWireguardConfigAndCacheIt()? Seems unnecessary to do all the time.

Reply:
Ah good call, this actually would allow me to move the `lastPrivateKeyUpdateTimeInMillis` inside the `Config`. Which is something I was thinking about doing instead of having to maintain it separately